### PR TITLE
Add AddDependent to the DependentService  

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#9267](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9267)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description
- Add AddDependent service

## Testing

* [X] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

 NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

**Review by sam-warren — DISMISSED** [review]

**Review by sslaws — CHANGES_REQUESTED** [review]
What about unit tests?

**Review by tiagograf-AOT — APPROVED** [review]

**Comment by leobcgov** [comment]
> What about unit tests?
Good question. How come this PR doesn't show the updated unit tests? It is here: https://github.com/bcgov/healthgateway/blob/features/9267/Apps/WebClient/test/unit/Services.Test/DependentService_Test.cs

**Review by tiagograf-AOT — APPROVED** [review]

File: Apps/WebClient/src/Server/Services/DependentService.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                || !patientResult.ResourcePayload.FirstName.Equals(addDependentRequest.FirstName, StringComparison.OrdinalIgnoreCase)
+                || patientResult.ResourcePayload.Birthdate.Year != addDependentRequest.DateOfBirth.Year
+                || patientResult.ResourcePayload.Birthdate.Month != addDependentRequest.DateOfBirth.Month
+                || patientResult.ResourcePayload.Birthdate.Day != addDependentRequest.DateOfBirth.Day
+                || !patientResult.ResourcePayload.Gender.Equals(addDependentRequest.Gender, StringComparison.OrdinalIgnoreCase))
+            {
+                return new RequestResult<DependentModel>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultError = new RequestResultError() { ResultMessage = "Information of the Dependent enterned don't match.", ErrorCode = ErrorTranslator.ServiceError(ErrorType.InvalidState, ServiceType.Patient) },
```
Comment (sslaws) [inline Apps/WebClient/src/Server/Services/DependentService.cs:86]: grammer and typo - please get wording from Nino

File: Apps/WebClient/src/Server/Services/DependentService.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            {
+                return new RequestResult<DependentModel>()
+                {
+                    ResultStatus = ResultType.Error,
+                    ResultError = new RequestResultError() { ResultMessage = "Communication Exception when trying to retrieve the Dependent", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Patient) },
+                };
+            }
+
+            // Verify dependent's details entered by user
+            if (!patientResult.ResourcePayload.LastName.Equals(addDependentRequest.LastName, StringComparison.OrdinalIgnoreCase)
```
Comment (sslaws) [inline Apps/WebClient/src/Server/Services/DependentService.cs:76]: Can you just add a comparison method to the AddDependentRequest to clean this up.

**Review by sslaws — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/Server/Services/DependentService.cs (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/src/Server/Services/DependentService.cs:86]: @ninosamson can you please provide the error message?

File: Apps/WebClient/src/Server/Services/DependentService.cs (hunk 2 above)
Comment (leobcgov) [reply Apps/WebClient/src/Server/Services/DependentService.cs:76]: @sslaws do you mean that having a method in AddDependentRequest accepting a PatientResourcePayload as the input?

Comment (leobcgov) [reply Apps/WebClient/src/Server/Services/DependentService.cs:76]: updated per our discussion by moving the checks to the AddDependentRequest model

File: Apps/WebClient/src/Server/Services/DependentService.cs (hunk 1 above)
Comment (leobcgov) [reply Apps/WebClient/src/Server/Services/DependentService.cs:86]: per @ninosamson, the err message should be "The information you entered did not match. Please try again."

**Review by sslaws — APPROVED** [review]

**Review by bradhead — APPROVED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**